components/AddChildForm.jsx

```diff
@@ -1,5 +1,206 @@
+`;
 
 export default function AddChildForm() {
-  return <form>AddChildForm</form>;
+  const router = useRouter();
+  const [inputData, setInputData] = useState({
+    name: "",
+    birth_date: "",
+    imgUrl: "",
+  });
+
+  const [isFormComplete, setFormComplete] = useState(false);
+
+  useEffect(() => {
+    if (
+      inputData.name.length !== 0 &&
+      inputData.birth_date.length !== 0 &&
+      inputData.imgUrl.length !== 0
+    ) {
+      setFormComplete(true);
+    } else {
+      setFormComplete(false);
+    }
+  }, [inputData]);
+
+  async function handleSubmit(e) {
+    e.preventDefault();
+
+    const formData = new FormData(e.target);
+    const childData = Object.fromEntries(formData);
+
+    const response = await fetch("/api/children_items", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(childData),
+    });
+    if (response.ok) {
+      mutate();
+    }
+    setInputData(childData);
+    router.push("/");
+  }
+
+  function handleChange(e) {
+    const key = e.target.name;
+    const value = e.target.value;
+
+    setInputData((inputData) => ({ ...inputData, [key]: value }));
+  }
+
+  const shortendUrl = inputData?.imgUrl.slice(0, 20) + "...";
+
+  return (
+    <>
+      <StyledForm onSubmit={handleSubmit}>
+        <StyledHeading>Add Child</StyledHeading>
+        <StyledInputContainer>
+          <StyledLabel htmlFor="name">Child Name</StyledLabel>
+          <StyledInput
+            name="name"
+            required
+            type="text"
+            id="name"
+            onChange={handleChange}
+            value={inputData.name}
+            placeholder={inputData.name ? inputData.name : "..."}
+          />
+        </StyledInputContainer>
+        <StyledInputContainer>
+          <StyledLabel htmlFor="birth_date">Date of Birth</StyledLabel>
+          <StyledDateInput
+            name="birth_date"
+            required
+            type="date"
+            id="birth_date"
+            onChange={handleChange}
+            value={inputData.birth_date}
+            placeholder={
+              inputData.birth_date ? inputData.birth_date : "01.01.2020"
+            }
+          />
+        </StyledInputContainer>
+        <StyledInputContainer>
+          <StyledLabel htmlFor="imgUrl">Image URL</StyledLabel>
+          <StyledInput
+            required
+            name="imgUrl"
+            type="text"
+            id="imgUrl"
+            onChange={handleChange}
+            value={inputData.imgUrl}
+            placeholder={inputData.imgUrl ? shortendUrl : "..."}
+          />
+        </StyledInputContainer>
+        <StyledBtnContainer>
+          {isFormComplete ? (
+            <>
+              <StyledLinkBtn href="/">Cancel</StyledLinkBtn>
+              <Button type="submit">Add</Button>
+            </>
+          ) : (
+            <StyledLinkBtn href="/">Cancel</StyledLinkBtn>
+          )}
+        </StyledBtnContainer>
```

Comment:
I think it would work just the same like this while being more readable:

```suggestion
       <StyledBtnContainer>
          <StyledLinkBtn href="/">Cancel</StyledLinkBtn>
         {isFormComplete && <Button type="submit">Add</Button>}
       </StyledBtnContainer>
```